### PR TITLE
Document jupyter notebook status and how to install a jupyter kernel

### PR DESCRIPTION
**Pull request by ngoldbaum**

### CODE REVIEW

File: docs/installing_cpython.md [hunk 1]
```diff
@@ -158,6 +158,20 @@ and installing the free-threaded binaries is also possible:
 
     On macOS, the Python framework built with the free-threading ABI can be found at `$(brew --prefix)/Frameworks/PythonT.framework`.
 
+## Installing a Jupyter Kernel
+
+While jupyter [does not currently support free-threaded
```
Comment (rgommers) [inline docs/installing_cpython.md:163]: nit: Jupyter with a capital J

File: docs/installing_cpython.md [hunk 2]
```diff
@@ -158,6 +158,20 @@ and installing the free-threaded binaries is also possible:
...
+Python](https://github.com/jupyter/notebook/issues/7512), you can use Jupyter
```
Comment (rgommers) [inline docs/installing_cpython.md:164]: If you're linking to issues, maybe Jupyter Lab needs one too? https://github.com/jupyter/notebook/issues/7512 is closest.

**Review by rgommers — COMMENTED** [review]
Thanks @ngoldbaum! This does seem useful. The content looks reasonable to me, but I haven't tried this at all yet and am not sure what (if anything) is happening regarding free-threading in Jupyter land. @minrk since you were working on `pyzmq` free-threading support, I suspect you may be able to tell us more here. Does this look fine to you to add?

File: docs/installing_cpython.md (hunk 2 above)
Comment (minrk) [reply docs/installing_cpython.md:164]: 👍 I think the jupyterlab one is a better canonical issue. We'll probably close the one on `jupyter/notebook` as a duplicate.

**Comment by minrk** [comment]
Yeah, I think this is fine. FWIW, I don't expect to need any patches in Jupyter. I added some [detail to the linked issue](https://github.com/jupyterlab/jupyterlab/issues/16915#issuecomment-2456867517) that we are mostly just waiting on dependencies, and an example of how I was able to make a free-threading env with JupyterLab.

**Comment by ngoldbaum** [comment]
Thanks for the extra detail! I'm actively working on Shipping PyO3 0.23 so hopefully after that I can help with unblocking Jupyter's dependencies.

**Pull request merged**